### PR TITLE
Filter notifications

### PR DESCRIPTION
**Pull request by khamui**

### CODE REVIEW

File: apps/userdashboard/assets/js/a4_candy_userdashboard/ModerationCommentList.jsx [hunk 1]
```diff
@@ -48,52 +49,53 @@ export default class ModerationCommentList extends Component {
...
-                  aiClassified
-                />
-              </li>
-            ))}
-          </ul>
+          {!isLoaded
+            ? (
+              <div className="d-flex justify-content-center">
+                <div className="spinner-border" role="status">
+                  <span className="sr-only">{loadingText}</span>
```
[the remark this answers is not in the document]
Comment (khamui) [reply apps/userdashboard/assets/js/a4_candy_userdashboard/ModerationCommentList.jsx:77]: ah nice. thanks for pointing there. i just picked a bootstrap spinner without doublechecking.

File: apps/userdashboard/assets/js/a4_candy_userdashboard/ModerationCommentList.jsx [hunk 2]
```diff
@@ -9,37 +9,38 @@ export default class ModerationCommentList extends Component {
...
-          userClassifications: json,
-          isLoaded: true
-        })
-      }).catch((err) => {
-        console.log(err)
-      })
+  filterChangeHandle (filter) {
+    this.setState({ filter: filter })
+    this.setState({ isLoaded: false })
+    this.loadData(filter)
```
[the remark this answers is not in the document]
Comment (khamui) [reply apps/userdashboard/assets/js/a4_candy_userdashboard/ModerationCommentList.jsx:26]: totally right! will change

File: .gitignore [hunk 3]
```diff
@@ -65,3 +65,6 @@ media/
 # test data
 *.key
 *.pem
+
+# prettier configs
+*prettier*
```
[the remark this answers is not in the document]
Comment (khamui) [reply .gitignore:70]: clarified in standup

File: apps/userdashboard/assets/js/a4_candy_userdashboard/FilterBar.jsx [hunk 4]
```diff
@@ -1,29 +1,49 @@
...
+    setCurrFilterItem(filterItem)
+    props.onFilterChange(filterItem.filter)
+  }
+
   return (
-    <div className="filter-bar justify-content-end" role="group" aria-label={django.gettext('Filter')}>
-      <div className="me-sm-1 mt-2 mt-sm-0">
+    <div
+      className="filter-bar justify-content-end"
+      role="group"
```
Comment (phillimorland) [inline apps/userdashboard/assets/js/a4_candy_userdashboard/FilterBar.jsx:25]: I don't think we should use `group` without using `groupitem`? but it's only just something i've learned about so it's possible we do it wrong elsewhere https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_group_role but semantically I don't think it makes sense

File: apps/userdashboard/assets/js/a4_candy_userdashboard/FilterBar.jsx [hunk 5]
```diff
@@ -1,29 +1,49 @@
...
-              <a className="selected" href="/#">all</a>
-            </li>
-            <li>
-              <a href="/#">pending</a>
-            </li>
-            <li>
-              <a href="/#">archived</a>
-            </li>
+            {filterItems.map((filterItem, i) => (
+              <li key={`${i}_${filterItem.name}`}>
```
Comment (phillimorland) [inline apps/userdashboard/assets/js/a4_candy_userdashboard/FilterBar.jsx:41]: this might be a super dumb question but why do you use `$` here?

**Review by phillimorland — APPROVED** [review]
Looking great, just one tiny thing and a question but once group is gone, feel free to merge! :)

File: apps/userdashboard/assets/js/a4_candy_userdashboard/FilterBar.jsx (hunk 5 above)
Comment (khamui) [reply apps/userdashboard/assets/js/a4_candy_userdashboard/FilterBar.jsx:41]: ah, i am using `template literals` here. meaning whatever is in between `${ .. }` will be treated as expression.
A better example would be:

`Hello ${name}, welcome to this.`

**Pull request merged**